### DISCUSSION TITLE
How to deal with custom schemas with postgresql?

### DISCUSSION THREAD

**Discussion by NilsIrl** [diesel-rs/diesel#2990] [Q&A] [answered]
I have the following migration:

```sql
CREATE SCHEMA myschema;

CREATE TABLE myschema.Person (
    PersonId bigint PRIMARY KEY,
    Name text NOT NULL,
    Code text NOT NULL,
    Token text NOT NULL
);
```

After running `diesel migration run` an empty `src/schema.rs` file is generated. I then tried again without using a custom schema (`myschema`) and it worked as expected. What can I do to make it work? This seems to have been implemented in https://github.com/diesel-rs/diesel/issues/447

**Comment by weiznich** [comment, marked as answer]
You need to configure diesel cli to use the right database schema to generate the corresponding schema.rs file. Checkout [this](https://diesel.rs/guides/configuring-diesel-cli.html) guide for details.

**Reply by NilsIrl** [reply]
Thanks a lot
